src/sqs/error-handlers.ts

```diff
@@ -0,0 +1,69 @@
+import {Context, ErrorHandler, ErrorHandlingResult} from "./index";
+import {JiraClientError} from "../jira/client/axios";
+import {Octokit} from "@octokit/rest";
+import {RateLimitingError} from "../config/enhance-octokit";
+import {emitWebhookFailedMetrics} from "../util/webhooks";
+import {PushQueueMessagePayload} from "./push";
+
+/**
+ * Sometimes we can get errors from Jira and GitHub which does not indicate a failured webhook. For example:
+ *  Jira site is gone, we'll get 404
+ *  GitHub App is not installed anymore we'll get 401
+ *  etc.
+ *
+ *  In such cases webhook processing doesn't make sense anymore and we need to silently discard these errors
+ */
+const UNRETRYABLE_STATUS_CODES = [401, 404, 403];
+
+const RATE_LIMITING_DELAY_BUFFER_SEC = 10;
+const EXPONENTIAL_BACKOFF_BASE_SEC = 60;
+const EXPONENTIAL_BACKOFF_MULTIPLIER = 3;
+
+export const jiraOctokitErrorHandler : ErrorHandler<any> = async (error: JiraClientError | Octokit.HookError | RateLimitingError | Error,
```

Comment:
This code is moved as it was from https://github.com/atlassian/github-for-jira/pull/945/files#diff-512ceb7a4811d8fa1e051dc4fe9ea670c609aab48242bfa97ae88398285a9ab8L43 to make it a "common" handler (because we are using same logic in both cases)